e2e/app/eoa/eoa.go

```diff
@@ -4,85 +4,128 @@ package eoa
-		TypeRelayer: common.HexToAddress("0x07804D7B8be635c0C68Cdf3E946114221B12f4F7"),
-	},
+	return nil
 }
 
-// MustAddress returns the address for the EOA identified by the network and type.
-func MustAddress(network netconf.ID, typ Type) common.Address {
-	resp, _ := Address(network, typ)
+// MustAddress returns the address for the EOA identified by the network and role.
+func MustAddress(network netconf.ID, role Role) common.Address {
+	resp, _ := Address(network, role)
 	return resp
 }
 
-// Address returns the address for the EOA identified by the network and type.
-func Address(network netconf.ID, typ Type) (common.Address, bool) {
-	if network == netconf.Devnet {
-		return crypto.PubkeyToAddress(devnetKeys[typ].PublicKey), true
+// Address returns the address for the EOA identified by the network and role.
+func Address(network netconf.ID, role Role) (common.Address, bool) {
+	accounts, ok := statics[network]
+	if !ok {
+		return common.Address{}, false
 	}
 
-	resp, ok := secureAddrs[network][typ]
+	for _, account := range accounts {
+		if account.Role == role {
+			return account.Address, true
+		}
+	}
 
-	return resp, ok
+	return common.Address{}, false
 }
 
-// PrivateKey returns the private key for the EOA identified by the network and type.
-func PrivateKey(ctx context.Context, network netconf.ID, typ Type) (*ecdsa.PrivateKey, error) {
-	if network == netconf.Devnet {
-		return devnetKeys[typ], nil
-	}
-
-	addr, ok := secureAddrs[network][typ]
+// PrivateKey returns the private key for the EOA identified by the network and role.
+func PrivateKey(ctx context.Context, network netconf.ID, role Role) (*ecdsa.PrivateKey, error) {
+	acc, ok := accountForRole(network, role)
 	if !ok {
-		return nil, errors.New("eoa key not defined", "network", network, "type", typ)
+		return nil, errors.New("eoa key not defined", "network", network, "role", role)
+	}
+	if network == netconf.Devnet {
+		return acc.PrivateKey, nil
 	}
 
-	k, err := key.Download(ctx, network, string(typ), key.EOA, addr.Hex())
+	k, err := key.Download(ctx, network, string(role), key.EOA, acc.Address.Hex())
 	if err != nil {
 		return nil, errors.Wrap(err, "download key")
 	}
 
 	return k.ECDSA()
 }
+
+// accountForRole returns the account for the network and role.
+func accountForRole(network netconf.ID, role Role) (Account, bool) {
+	accounts, ok := statics[network]
+	if !ok {
+		return Account{}, false
+	}
+	for _, account := range accounts {
+		if account.Role == role {
+			return account, true
+		}
+	}
+
+	return Account{}, false
+}
+
+func MustAddressesForRoles(network netconf.ID, roles ...Role) []common.Address {
+	accounts := statics[network]
+	var addresses []common.Address
+	for _, role := range roles {
+		for _, account := range accounts {
+			if account.Role == role {
+				addresses = append(addresses, account.Address)
+			}
+		}
+	}
+
+	return addresses
+}
+
+func addr(hex string) common.Address {
```

Comment:
rather use common.HexToAddress(hex) directly, this abbreviation isn't super clear